lib/decidim/term_customizer/context/controller_context.rb

```diff
@@ -27,6 +27,9 @@ def resolve!
               # we take this into account here.
             end
           end
+
+          @space = controller.send(:set_group) if controller.respond_to?(:set_group, true)
```

Comment:
Elegant, but also risky.

This will break if any decidim controller, module controller or userland controller has a `set_group` method, which is not an unusual method name. There is already a model `Decidim::UserGroup` with a `Decidim::GroupsController`, which could easily add a `set_group` method and nobody would suspect that the term customizer will break after that change.

It will also break if decidim changes their internal implementation. `ParticipatoryProcessGroupsController#set_group` is [not a public method](https://github.com/decidim/decidim/blob/f8f1eac79fb4e8890c8a637a28067ff5d73487fd/decidim-participatory_processes/app/controllers/decidim/participatory_processes/participatory_process_groups_controller.rb#L31..L33) and could be renamed or removed or change behaviour even on minor or patch releases of decidim.

Also, you are putting a ProcessGroup into a variable named `@space`, even though a ProcessGroup is not a participatory space. Maybe this works right now, but it might break if the term customizer ever relies on some features of a participatory space other than only the id. In typed languages, this would be completely disallowed by the compiler, and it looks like an anti-pattern to me.

It would be better (although more complicated) to keep the ProcessGroup in a separate variable and add a separate query extension for it, as I suggested in our technical concept in https://github.com/puzzle/decidim-zuerich/issues/327#issuecomment-1303559843

Reply:
Thanks a lot, i had the intuition that it wasn't the right way but i didn't knew why.
It should be better now !